src/parser/Parser.ts

```diff
@@ -412,6 +412,11 @@ export class Parser {
                 });
                 //consume the statement separator
                 this.consumeStatementSeparators();
+            } else if (this.peek().kind !== TokenKind.Identifier && !this.checkAny(...DeclarableTypes, ...AllowedTypeIdentifiers)) {
```

Comment:
Only try to find a type if the next token can be a type identifier -- basically, don't throw an exception if there isn't the correct kind of token next.
This is important so that we can do completions inside this expression